web/src/pages/Setting/Setting.tsx

```diff
@@ -62,7 +62,15 @@ const Setting: React.FC = () => {
         });
         setTimeout(() => {
           const redirect = getUrlQuery('redirect');
-          window.location.href = redirect ? decodeURIComponent(redirect) : '/';
+          const currentHost = window.location.host;
+          if (redirect) {
+            const redirectUrl = decodeURIComponent(redirect);
+            const redirectHost = redirectUrl.split('/')[2];
+            if (currentHost === redirectHost) {
+              window.location.href = redirectUrl;
```

Comment:
> To guard against untrusted URL redirection, it is advisable to avoid putting user input directly into a redirect URL. Instead, maintain a list of authorized redirects on the server; then choose from that list based on the user input provided.

After read the recommended note from codeql, it seems that all urls derived from user input are considered risky, so only limit the same host is not enough(I seem the risky is still exit). 

Since the redirectUrl could not come from server(the recommended way), we may try to use `history.push` instead of `window.location.href` to avoid this sec risky, which can also complete the page redirect action.

Reply:
I studied the source code and understood that `history.push()` is done via importing history from umi. I referred the [umi documentation on `history.push()`](https://umijs.org/api#history). 

So, in line 70, can we try to replace the code like below:
`history.push(redirectUrl);`

Reply:
> I studied the source code and understood that `history.push()` is done via importing history from umi. I referred the [umi documentation on `history.push()`](https://umijs.org/api#history).
> 
> So, in line 70, can we try to replace the code like below:
> `history.push(redirectUrl);`

you would better only pass the path in the `redirecrUrl` to the `history.push()`
so you have to：
1. extract the `path` form `redirecrUrl`
2. replace the code in line 70 to `history.push(path);`

just try it ^_^  @ThanoshanMV